From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mariell Hoversholm <proximyst@proximyst.com>
Date: Mon, 29 Jun 2020 22:30:33 +0200
Subject: [PATCH] Revert Improve Legacy Component serialization size

This reverts commit 4e2ac9cf3c6f985098e5c6f52e98a08a483f8384.

This creates several regressions in the API.
---
 .../craftbukkit/util/CraftChatMessage.java    | 23 +------------------
 1 file changed, 1 insertion(+), 22 deletions(-)

diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftChatMessage.java b/src/main/java/org/bukkit/craftbukkit/util/CraftChatMessage.java
index 54f6f7d86741c3f318fa76bafe04b55cf420c35b..a423e3bf3906d3114af885c48d0b45d88d7e7d8b 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/CraftChatMessage.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/CraftChatMessage.java
@@ -45,7 +45,6 @@ public final class CraftChatMessage {
         // Separate pattern with no group 3, new lines are part of previous string
         private static final Pattern INCREMENTAL_PATTERN_KEEP_NEWLINES = Pattern.compile("(" + String.valueOf(org.bukkit.ChatColor.COLOR_CHAR) + "[0-9a-fk-orx])|((?:(?:https?):\\/\\/)?(?:[-\\w_\\.]{2,}\\.[a-z]{2,4}.*?(?=[\\.\\?!,;:]?(?:[" + String.valueOf(org.bukkit.ChatColor.COLOR_CHAR) + " ]|$))))", Pattern.CASE_INSENSITIVE);
         // ChatColor.b does not explicitly reset, its more of empty
-        private static final ChatModifier EMPTY = ChatModifier.b.setItalic(false); // Paper - OBFHELPER
         private static final ChatModifier RESET = ChatModifier.b.setBold(false).setItalic(false).setUnderline(false).setStrikethrough(false).setRandom(false);
 
         private final List<IChatBaseComponent> list = new ArrayList<IChatBaseComponent>();
@@ -67,7 +66,6 @@ public final class CraftChatMessage {
             Matcher matcher = (keepNewlines ? INCREMENTAL_PATTERN_KEEP_NEWLINES : INCREMENTAL_PATTERN).matcher(message);
             String match = null;
             boolean needsAdd = false;
-            boolean hasReset = false; // Paper
             while (matcher.find()) {
                 int groupId = 0;
                 while ((match = matcher.group(++groupId)) == null) {
@@ -113,26 +111,7 @@ public final class CraftChatMessage {
                             throw new AssertionError("Unexpected message format");
                         }
                     } else { // Color resets formatting
-                        // Paper start - improve legacy formatting
-                        ChatModifier previous = modifier;
-                        modifier = (!hasReset ? RESET : EMPTY).setColor(format);
-                        hasReset = true;
-                        if (previous.isBold()) {
-                            modifier = modifier.setBold(false);
-                        }
-                        if (previous.isItalic()) {
-                            modifier = modifier.setItalic(false);
-                        }
-                        if (previous.isRandom()) {
-                            modifier = modifier.setRandom(false);
-                        }
-                        if (previous.isStrikethrough()) {
-                            modifier = modifier.setStrikethrough(false);
-                        }
-                        if (previous.isUnderlined()) {
-                            modifier = modifier.setUnderline(false);
-                        }
-                        // Paper end
+                        modifier = RESET.setColor(format);
                     }
                     needsAdd = true;
                     break;
